web/src/main/java/com/bazaarvoice/emodb/web/scanner/rangescan/LocalRangeScanUploader.java

```diff
@@ -243,7 +252,16 @@ public void run() {
             int partCountForFirstShard = 1;
             Batch batch = new Batch(context, partCountForFirstShard);
 
-            Iterator<MultiTableScanResult> allResults = _dataTools.multiTableScan(multiTableScanOptions, tableSet, LimitCounter.max(), ReadConsistency.STRONG, null);
+            // check if there is a stash cut off time.
+            Map<String, StashRunTimeInfo> stashTimeInfoMap = _compactionControlSource.getStashTimesForPlacement(placement);
+            DateTime cutoffTime = null;
+            if (stashTimeInfoMap.size() > 0) {
+                cutoffTime = new DateTime(stashTimeInfoMap.entrySet().stream()
+                        .min((entry1, entry2) -> entry1.getValue().getTimestamp() > entry2.getValue().getTimestamp() ? 1 : -1)
+                        .get().getValue().getTimestamp());
+            }
```

Comment:
I don't think this is the right way to check for the cutoff time.  I think it make more sense to store it with the stash state and pass it as a parameter here.  The risk is that if multiple stashes are running concurrently this will use the minimum start time of both stashes, which is incorrect if they didn't start at the same time.  This is rare but not impossible, so better to rely on the known start time for this stash rather than the control source artifact.